src/XGrid.cu

```diff
@@ -0,0 +1,260 @@
+/* 
+
+Implements the grid-multiplication onto a GPU using CUDA. 
+
+*/
+#include <iostream>
+#include "bifrost/XGrid.h"
+#include "assert.hpp"
+#include "trace.hpp"
+#include "utils.hpp"
+#include "cuda.hpp"
+#include "cuda/stream.hpp"
+#include "Complex.hpp"
+
+struct __attribute__((aligned(1))) nibble2 {
+    // Yikes!  This is dicey since the packing order is implementation dependent!  
+    signed char y:4, x:4;
+};
+
+struct __attribute__((aligned(1))) blenib2 {
+    // Yikes!  This is dicey since the packing order is implementation dependent!
+    signed char x:4, y:4;
+};
+
+template<typename In, typename Out>
+__global__ void XGrid_kernel(int npol, int gridsize, int nbatch, int nchan,
```

Comment:
It doesn't appear that `gridsize`, `nbatch`, or `nchan` are used

Reply:
Yes they do not have use in the final grid multiplication but having or removing them doesn't change or affect performance in anyway. These can be useful while performing checks (I was using them) at times to know if I'm accessing the right memory location etc..

Reply:
Unused variable arguments have been eliminated.